src/infoview.ts

```diff
@@ -120,12 +126,17 @@ export class InfoProvider implements Disposable {
                     payload: JSON.stringify(e)
                 })
             ),
-            this.proxyConnection.jsonMessage.on(e =>
+            this.proxyConnection.jsonMessage.on(e => {
+                if ('record' in e && 'state' in e.record && 'widget' in e.record){
```

Comment:
This will trigger on every hover as well (i.e., when you move the mouse around).  A better place to intercept would be here: https://github.com/leanprover/vscode-lean/blob/2b43982c4c6305a0f20f156152a60613a6f1a683/infoview/info.tsx#L173
You'd then need to send a message from the webview back to the extension like this: https://github.com/leanprover/vscode-lean/blob/2b43982c4c6305a0f20f156152a60613a6f1a683/infoview/server.ts#L25-L27
And receive the message here:
https://github.com/leanprover/vscode-lean/blob/2b43982c4c6305a0f20f156152a60613a6f1a683/src/infoview.ts#L195